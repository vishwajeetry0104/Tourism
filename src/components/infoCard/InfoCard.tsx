import React from 'react';
import { View, Text, Image } from 'react-native';
import Warning from '../../assets/svgIcons/Warning.svg';
import InfoCardStyles from './InfoCardStyles';

type Props = {
    headingTitle: string,
    message: string,
}

const InfoCard = (props: Props) => {
    const {headingTitle, message} = props;
  return (
    <View style={InfoCardStyles.container}>
      <Warning height={48} width={48} fill="#FF5900" />
      <Text style={InfoCardStyles.headingTitle}>{headingTitle}</Text>
      <Text style={InfoCardStyles.message}>{message}</Text>
    </View>
  );
}

export default InfoCard;
