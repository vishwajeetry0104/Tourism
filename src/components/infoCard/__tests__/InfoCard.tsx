import React from 'react';
import {act, create} from 'react-test-renderer';
import InfoCard from '../InfoCard';

describe('InfoCard', () => {
  const props = {
    headingTitle: 'Testing title',
    message: 'Nothing to display'
  };

  const createWrapper = (updatedProps = props) => {
    return create(<InfoCard {...updatedProps} />);
  };

  let defaultWrapper;

  act(() => {
    defaultWrapper = createWrapper();
  });

  afterAll(() => {
    defaultWrapper.unmount();
  });

  it('should render InfoCard component without issues', () => {
    expect(defaultWrapper).toMatchSnapshot();
  });
});
