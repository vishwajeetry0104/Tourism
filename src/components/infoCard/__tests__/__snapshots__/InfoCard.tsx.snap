// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoCard should render InfoCard component without issues 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
      "marginTop": 48,
      "paddingHorizontal": 16,
    }
  }
>
  <SvgMock
    fill="#FF5900"
    height={48}
    width={48}
  />
  <Text
    style={
      {
        "color": "#FFFFFF",
        "fontSize": 24,
        "fontWeight": "700",
        "letterSpacing": 0.24,
        "paddingBottom": 8,
      }
    }
  >
    Testing title
  </Text>
  <Text
    style={
      {
        "color": "#FFFFFF",
        "fontSize": 16,
        "fontWeight": "700",
        "letterSpacing": 0.24,
        "paddingBottom": 8,
      }
    }
  >
    Nothing to display
  </Text>
</View>
`;
